tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,307 @@
+    try {
+        $psData    = $db->pselect($dataQuery, array());
+    } catch (DatabaseException $e) {
+        print_r($errors);
+        echo "\nResolve errors and run script again.\n";
+        die();
+    }
+    foreach ($psData as $entry) {
+        $status     = $entry[$consentName];
+        $date       = $entry[$consentDate];
+        $withdrawal = $entry[$consentWithdrawal];
+        $consentID  = $entry['ID'];
+        $candID     = $entry['CandID'];
+        // Check if consent status is not_answered
+        if($status === "not_answered") {
+            array_push($errors, "Deprecated consent status 'not_answered' found for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       Please change to a valid status or remove data.");
+        }
+        // Check if consent status is null or empty but there are dates
+        if(empty($status) && (!empty($date) || !empty($withdrawal))) {
+            array_push($errors, "Consent status not found for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . " where there are dates.");
+        }
+        // Check if consent status is yes but consent date is empty
+        if($status === "yes" && empty($date)) {
+            array_push($errors, "The date of consent is missing for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       All 'yes' statuses must have a consent date.");
+        }
+        // Check if consent status is no and consent date given, but withdrawal date is empty
+        if($status === "no" && !empty($date)) {
+            if(empty($withdrawal)) {
+                array_push($errors, "The date of withdrawal is missing for " . $consentName . ":
+                           [ID]     => " . $consentID . "
+                           [CandID] => " . $candID . "
+                           All 'no' statuses with given consent date must have a withdrawal date.");
+            }
+        }
+        // Check for zero dates
+        if($date === "0000-00-00" || $withdrawal === "0000-00-00") {
+            array_push($errors, "Zero dates found in participant_status for " . $consentName . ":
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       Please remove date or run /tools/DB_date_zeros_removal.php.");
+        }
+    }
+}
+////////////////////////////////////////////////
+// CHECK NEW TABLES AND CONFIG SETTINGS EXIST //
+////////////////////////////////////////////////
+
+// Check useConsent was inserted into ConfigSettings
+try {
+    $configID = $db->pselectOne("SELECT ID FROM ConfigSettings WHERE Name='useConsent'", array());
+} catch (DatabaseException $e) {
```

Comment:
Why would this throw a database exception? If a simple query with no parameters throws an exception it really is an exceptional circumstance and should probably not be caught.

Reply:
(one the other hand, it *should* check if the value is empty...)

Reply:
you are right, i should fix this